gtsam/navigation/tests/testCombinedImuFactor.cpp

```diff
@@ -316,6 +317,38 @@ TEST(CombinedImuFactor, Accelerating) {
   EXPECT(assert_equal(estimatedCov, expected, 0.1));
 }
 
+/* ************************************************************************* */
+TEST(CombinedImuFactor, ResetIntegration) {
+  const double a = 0.2, v = 50;
+
+  // Set up body pointing towards y axis, and start at 10,20,0 with velocity
+  // going in X The body itself has Z axis pointing down
+  const Rot3 nRb(Point3(0, 1, 0), Point3(1, 0, 0), Point3(0, 0, -1));
+  const Point3 initial_position(10, 20, 0);
+  const Vector3 initial_velocity(v, 0, 0);
+
+  const AcceleratingScenario scenario(nRb, initial_position, initial_velocity,
+                                      Vector3(a, 0, 0));
+
+  const double T = 3.0;  // seconds
+
+  CombinedScenarioRunner runner(scenario, testing::Params(), T / 10);
+
+  PreintegratedCombinedMeasurements pim = runner.integrate(T);
+  // Make copy for testing different conditions
+  PreintegratedCombinedMeasurements pim2 = pim;
+
+  // Test default method
+  pim.resetIntegration();
+  Matrix6 expected = Z_6x6;
```

Comment:
Do we expect zero? I thought you were setting it to `preintMeasCov_.block<6, 6>(9, 9);` ?

Reply:
Okay made a bunch of updates to enable this. I actually like the added flexibility.